dpcpp/components/atomic.dp.hpp

```diff
@@ -203,6 +279,14 @@ __dpct_inline__ T atomic_add(T *__restrict__ addr, T val)
 }
 
 
+template <cl::sycl::access::address_space addressSpace = atomic::global_space,
+          typename T>
+__dpct_inline__ T atomic_max(T *__restrict__ addr, T val)
```

Comment:
I am not certain `__restrict__` makes any difference here, since the function takes only a single pointer parameter.

Reply:
I would keep it as it can also improve caching.